packages/ember-application/lib/system/application-instance.js

```diff
@@ -83,24 +79,20 @@ let ApplicationInstance = EmberObject.extend(RegistryProxy, ContainerProxy, {
   rootElement: null,
 
   init() {
-    this._super(...arguments);
+    let application = get(this, 'application');
```

Comment:
Good catch, @mixonic. I was able to refactor it such that the engine instance is aware that `application` might be set instead of `base` (i.e. if `base` is not set, then `application` is checked). if others agree that `base` is general enough to specify the base class for both engine and application instances, we can consider deprecating the `application` property at some point.